Model/Quote/SaveOrderBeforeSalesModelQuoteObserver.php

```diff
@@ -95,38 +100,44 @@ public function execute(\Magento\Framework\Event\Observer $observer)
     }
 
     /**
-     * @param $quote
+     * @param int $quote
      *
      * @return bool
      */
-    private function isMyParcelMethod($quote) {
+    private function isMyParcelMethod(int $quote): ?bool
+    {
         $myParcelMethods = array_keys(Carrier::getMethods());
         $shippingMethod  = $quote->getShippingAddress()->getShippingMethod();
 
-        if ($this->array_like($shippingMethod, $myParcelMethods)) {
+        if ($this->isMyparcelRelated($shippingMethod, $myParcelMethods)) {
             return true;
         }
 
-        if ($this->array_like($shippingMethod, $this->parentMethods)) {
+        if ($this->isMyparcelRelated($shippingMethod, $this->parentMethods)) {
             return true;
         }
 
         return false;
     }
 
     /**
-     * @param $input
-     * @param $data
+     * @param int|null $input
+     * @param array    $data
      *
      * @return bool
      */
-    private function array_like($input, $data) {
-        $result = array_filter($data, function ($item) use ($input) {
-            if (stripos($input, $item) !== false) {
-                return true;
+    private function isMyparcelRelated(?int $input, array $data): ?bool
```

Comment:
Hoe kan $input null zijn? En je returnt altijd `bool`, dus geen `?bool`. Fix dan ook nog even de casing in de functie naam :P